nifpga/session.py

```diff
@@ -456,226 +449,214 @@ def __init__(self,
         super(_FxpRegister, self).__init__(session,
                                            nifpga,
                                            bitfile_register,
-                                           base_address_on_device)
-
+                                           base_address_on_device,
+                                           read_func=nifpga["ReadArray%s" % DataType.U32],
+                                           write_func=nifpga["WriteArray%s" % DataType.U32])
         self._word_length = bitfile_register.word_length
         self._integer_word_length = bitfile_register.integer_word_length
-        self._radix_point = self._calculate_radix_point()
         self._overflow_enabled = bitfile_register.overflow
         self._signed = bitfile_register.signed
-        self._delta = self._calculate_delta(bitfile_register.word_length,
-                                            bitfile_register.integer_word_length)
+        self._delta = self._calculate_delta()
+        self._minimum = self._calculate_minimum()
+        self._maximum = self._calculate_maximum()
+        self._transfer_len = self._calculate_transfer_len()
+        self._ctype_type = self._ctype_type * self._transfer_len
+
+    def _calculate_delta(self):
+        """ The value persisted in the bitfile for delta is not always
+        correct, therefore we must calculate it manually.
+        """
+        return Decimal(2**(self._integer_word_length - self._word_length))
 
-    @property
-    def overflow(self):
-        """ This property should only set be whenever this register has overflow
-        enabled"""
-        return self._overflow
+    def _calculate_minimum(self):
+        """ The value persisted in the bitfile for the minimum value is
+        not always accurate, therefore we must calculate it manually.
+        """
+        if self._signed:
+            magnitude_bits = self._word_length - 1
+            return -1 * (2**(magnitude_bits) * self._delta)
+        else:
+            return 0
```

Comment:
Delta is specified to be Decimal in order not to end up with a rounded value. Here it would be overkill to return a Decimal with the value of zero, and because it is python we don't need to.